Выполнение задания в файле README

dmitrii-obukhov opened the pull request:
Домашнее задание по второй лекции.

dmitrii-obukhov committed:
Выполнение задания в файле README

Домашнее задание по второй лекции.

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
+    Gmail - почтовый сервис от компании Google. 
+    Предоставляет услуги по пересылке, получению и хранению электронных сообщений.
+    
+    Предполагаю, что авторизация на сайте выполнена.
```
авторизация, регистрация - важный функционал, неотъемлемая часть приложения, которое ты тестируешь. Нельзя вот так выбросить из поля зрения. Надо хоть по минимуму проверить. Так что внеси в майнд-мап по функционалу

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
+      5.4. Заголовок сообщения
+      5.5. Статус, показывающий наличие вложенных файлов
+      5.6. Дата получения сообщения
+    6. Настройки профиля - справа сверху
```
Можешь по-подробнее расписать, что входит в каждый пункт настроек? Еще на один уровень вниз распиши их, пожалуйста

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
+    4. Фильтры сообщений - по-центру
+      4.1. Фильтры, отвечающие за выделение сообщений
+      4.2. Фильтры, отвечающие за выбор группы (несорированные, соцсети, и т.д.)
+    5. Сообщения - по-центру, под фильтрами
```
Тут ты описываешь только отображение списка полученных сообщений. На мой взгляд есть 2 функции, касающиеся сообщений: отображение (в списке - полученные, отправленные, + фильтрация списков, отдельно открытое сообщение) и редактирование (новое сообщение, ответ на сообщение). Что если учесть это в майнд-мапе? Ты покроешь больше функционала

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
+      7.1. Информация об использованном пространстве
+      7.2. Условия, конфиденциальность
+      7.3. Дополнительная информация
 
```
Еще пропущена важная фича отправки сообщений с разными настройками: одну\нескольким ардесатам, с разными уровнями важности, с подтверждением получения, обработка ошибочных ситуаций - адресат не существует, например.

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
+          Используем технику разделения на классы: введенное поле не пустая строка | введенное поле пустая строка.
+          Делаем тест соответсвующий каждому из классов.
+          
+          - Поле сообщение.
```
Если на это поле не накладывается ограниений, это не значит, что его не надо тестировать. Наоборот, это значит, что на это поле существует вагон тестовых случаев и все их система должна корректно обрабатывать. Написание сообщения - это почти что самый важный функционал в почтовике (еще один - получение сообщений).
Будет не очень хорошо, если пользователь не сможет ввести больше 10 строк в сообщение, прикрепить файл, отправить кликабельную ссылку и тд. Есть форматирование текста. Можно заполнять не только поле адресата, но и копию и скрытую копию. Протестируй этот функционал тщательнее

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
+      7. Подблок "Ещё":
+        Проверяем, что при выборе соответсвующего подпункта этого меню, отображаются только сообщения, удовлетворяющие из соответсвующего раздела.
+        
+      Для пунктов 2 - 7 сделать можно по 2 теста: для случая, когда в выборке сообщения есть, и когда их нет.
```
ты взял для подробного тестирования весь блок меню, так что все тесты должны быть в тест-рейле. Занеси туда 2-7 тоже. Представь, что ты пишешь тестовый набор для другого тестировщика. Вообще, заноси как можно больше информации, которую ты узнал и проанализировал, в виде тестов или тестовой документации, для того, чтобы твоим коллегам не пришлось делать работу, которую ты уже проделал

nburunova replied:
если ты тестируешь только формирование и отображение выборки можно еще добавить тест - когда сообщений, больше чем на страницу, проверяем что работает пейджинг.
Итого на каждый пункт у тебя будет по 4 теста, например, для кругов:
1. Проверяем, что при выборе соответсвующего подпункта этого меню, отображаются только переписки с людьми из выбранных кругов. (самый важный тест)
2. в выборке нет сообщений
3. в выборке есть сообщения, малое количество, в пределах 1 страницы
4. в выборке много сообщений, верно работает пейджинг

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
+          Граница между классами: строка1, или строка2, или строка3 - пустая строка.
+          Делаем тест соответсвующий каждому из классов и грацице (дополнительно можно сделать отдельно тесты, когда из символов @ и . присутствует только один).
+          
+          Используем анализ причинно-следственных связей
```
Я пыталась донести, что в результате причинно-следственного анализа мы выясняем все возможные состояния системы. На мой взгляд, для "Написать сообщение" такие возможные состояние - редактирование сообщения, сообщение отправлено, сообщение сохранено в черновики, отмена сообщения,  состояние ошибки когда неверно заполнены поля. Можно еще какие-то состояние выделить. И есть ряд причин для каждого состояния. Для некторых состояний достаточно проверить, что они достижимы, для других можно провести более детальные проверки. Одни относятся к формированию сообщения, другие - к процессу отправки. Доработай тест-кейсы с учетом этого?

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
+      6. Круги
+      7. Ещё
+    Тестируем каждый из подблоков по-отдельности.
+      1. Подблок "Написать":
```
Тут важно понимать, что ты провеяешь, только формирование сообщения (редактирование) или отправку тоже?

nburunova commented:
по логике и понимаю сути - неплохо :+1:  , даже хорошо. Но по покрытию - не очень. Проведя только те тесты, что ты описал, я бы не решилась делать выводы, работает ли блок Меню или нет.

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
-## Итого с вас требуется:
-1. MindMap с декомпозицией функционала сайта и с подробной декомпозицией выбранного блока функционала (можно приложить файл или дать ссылку, если майнд-мап создан в онлайн-сервисе)
-2. Набор тест-кейсов в Тест-Рейле
+    Блок сообщения разбит на следующие подблоки
```
По сути меню - это доступ к нескольким функциональностям - я вижу тут такие как: 1. написание нового сообщения. 2. доступ к разным фильтрованным спискам сообщений (фильтрация + представление\отображение) - помеченные, важные, отправленные и т.д. Кроме того, можно создавать свои папки, которые наполняются в соотвествии с определнными параметрами фильтрации - а это тоже целая фича, кастомные папки и фильтры

nburunova commented on README.md:
```diff
@@ -1,39 +1,93 @@
+      4. Отравленные
+      5. Черновики
+      6. Круги
+      7. Ещё
```
под Еще скрывается еще много чего интересного, надо раскрыть и расписать их

dmitrii-obukhov committed:
Update README.md

dmitrii-obukhov committed:
Выполнение задания в файле README

Выполнил изменения в домашнем задании по 2 лекции. MindMap и тесты в тестРейле также изменены. В качестве тестируемого блока взял всего лишь подпункт Меню -> Написать сообщение, но постарался провести анализ подробно.

nburunova commented on README.md:
```diff
@@ -1,39 +1,160 @@
+          
+        1.4. Вложенный файлы
+          Для вложенных файлов необходимо протестировать поведение при загрузки:
+          - файлов меньшего размера, чем максимально допустимый;
```
надо еще граничное значение: размер файла близок к верхней границе, но не превышает ее

dmitrii-obukhov committed:
Выполнение задания в файле README

Домашнее задание по 2 лекции.
Добавлен ещё один тест, когда размер файла близок к верхней границе, но не превышает её.

nburunova commented:
1. Тесты на домены нижнего и верхнего уровня перепутаны - в тестовых данных на отсутствие домена верхнего уровня отсутствует нижний и наоборот
2. Набор тестов для проверки корректности введенного имени пользователя (10) - название блока не соответствует содержимому, в тестах у тебя написано "имя получателя", но вообще это имейл получителя, а не имя :)

nburunova commented:
я бы еще протестировала не только размер вложений, но и тип

nburunova commented:
https://projectt2015.testrail.net/index.php?/cases/view/12 и https://projectt2015.testrail.net/index.php?/cases/view/32 дублируют друг друга, и еще тесты про случай, когда тема не указана. Дублирования в тестах быть не должно. Еще тесты про гиперссылку дублируются

nburunova commented:
Набор тестов проверки отправления и получения сообщений (7) - блок заявлен как проверки отправления и получения, но я вижу тесты только про отправление. про получение нет ничего

dmitrii-obukhov commented:
@nburunova Исправил тесты на домены нижнего и верхнего уровня, исправил неудачные названия блоков, убрал дублирующие тесты, добавил тесты на различные типы вложений. Все изменения на тестрейле.

nburunova commented:
1. я бы еще проверяла макс длину темы (очень иного символов)
2. как думаешь, стоит проверять разные форматы для типов вложений? например, разные форматы изображений
3. сохранение черновиков - ты сделал акцент на теме сообщения, но стоит еще отдельно проверять, что сохраняется весь текст и форматирование текста, вложения, адресаты, если были введены
4. тесты на отправку сообщений - мне кажется еще нужно проверить, что получатель видит письмо так, как его отформатировал отправитель :)

Интересно, ты достаточно много всего рассмотрел в блоке про формирование сообщения, почему не перенёс это на черновики и отправку?

dmitrii-obukhov commented:
@nburunova 
1. Добавил тест на максимальную длину темы (граничные значения - когда символов максимальное число, и когда символов на 1 больше).
2. Перебирать все форматы это слишком затратный вариант по времени. Своим набором тестов я хотел показать, что разбил возможные типы вложений на классы эквивалентности и взял по одному элементу из каждого класса.
3. За формирование сообщений отвечает "Набор тестов для проверки формирования сообщений". "Набор тестов для проверки функции сохранения в черновиках" отвечает за само сохранение в черновиках, а тесты с темой и без темы нужны потому что в случае без темы вместо название темы подставляется <без темы> (эта особенность в  "Набор тестов для проверки формирования сообщений" не тестировалась). Если добавлять другие тесты, то они будут просто дублировать предыдущие.
4. Мне кажется, что тесты на получение сообщений это уже отдельный набор тестов. Здесь надо рассмотреть различные почтовые сервисы (возможно какие-то из них не поддерживают форматирования mail.google), что выходит за рамки этого задания.

То что рассмотрел в блоке про формирование сообщения относится и к черновикам, и к отправке сообщений, поэтому этот блок и выделен в отдельный, чтобы не рассматривать одни и те же вещи два раза.

nburunova commented:
"2 ". это долго, да. Но стоит проверить разок самые популярные. Или рассмотреть форматы, с которыми бывают проблемы (тут надо гуглить и спрашивать разработчиков)
И еще я подумала, что можно было проверить, как система реагирует на прикрепление какиз-нибудь исполняемых файлов, тех же ехе. Помню, я натыкалась на почтовики, которые отказывались прикреплять такие вложения

"3-4". ты пытаешься избежать дублирования в тестах - это правильно. Но согласись, что сформировав какое-то сложноотформатированное сообщение, хочется проверить, что оно верно сохранится в черновиках.  В случае, когда нужно проверить одно и то же в разных местах, я делаю так: а) в наборе тестов в коментарии указываю, что нужно проверить в блоке А и в блоке Б (для тебя это формирование сообщений и сохранение черновиков), или б) делаю несколько крупных тестов, которые включают в себя много проверок за раз и провожу эти тесты на функционале, который более вторичен. Для твоего случая: составляю письмо с интересной темой, получателями и сложным содержанием и  проверяю, что такое письмо сохраняется в черновике. 
С гугл почтой я бы провела вариант б) . Но совсем не проверять, как у меня сохраняются сложные сообщения в черновиках, я бы побоялась. Потому что в жизни много неожиданных багов, а пользователи жестоки

"4 ". про то, что ты не хочешь писать тесты на получение сообщений в этом блоке - ок. Но главное помни, что не в домашке, а в реальном проекте это нужно будет проверить

dmitrii-obukhov committed:
Обновление домашнего задания в файле README

Немного переосмыслил тесты. 
Теперь есть группа тестов, которые нужно проверить и при отправлении сообщений, и при сохранении в черновиках. И есть группа тестов, которые нужно проверить только при отправлении.
Ещё добавил тест с проверкой вложения формата ".exe" и подредактировал старые тесты на тестрейле.

dmitrii-obukhov commented:
@nburunova 
Внес изменения.

nburunova commented:
принято
